> @Pepo48 Could you please check if the following changes work [pedroigor@4f9736a](https://github.com/pedroigor/keycloak/commit/4f9736aee6ff50e16b3ef608170ecf7e83538cf1).

Now it throws NoSuchElementException:
```[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.9.0.Final:build (default) on project keycloak-quarkus-server-app: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#configureHibernate threw an exception: java.util.NoSuchElementException: No value present
[ERROR]         at java.util.Optional.get(Optional.java:135)
[ERROR]         at org.keycloak.quarkus.deployment.KeycloakProcessor.configureHibernate(KeycloakProcessor.java:99)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:936)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR]         at java.lang.Thread.run(Thread.java:748)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] -> [Help 1]```

![obrázok](https://user-images.githubusercontent.com/12138171/97590382-d3c19980-19fe-11eb-9f32-a15492a8f4a4.png)

--- a/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
+++ b/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
@@ -73,13 +73,13 @@ class KeycloakProcessor {
 
     /**
      * <p>Configures the persistence unit for Quarkus.
-     * 
-     * <p>The main reason we have this build step is because we re-use the same persistence unit from {@code keycloak-model-jpa} 
+     *
+     * <p>The main reason we have this build step is because we re-use the same persistence unit from {@code keycloak-model-jpa}
      * module, the same used by the Wildfly distribution. The {@code hibernate-orm} extension expects that the dialect is statically
      * set to the persistence unit if there is any from the classpath and we use this method to obtain the dialect from the configuration
      * file so that we can build the application with whatever dialect we want. In addition to the dialect, we should also be 
      * allowed to set any additional defaults that we think that makes sense.
-     * 
+     *
      * @param recorder
      * @param config
      * @param descriptors
@@ -88,7 +88,7 @@ class KeycloakProcessor {
     @BuildStep
     void configureHibernate(KeycloakRecorder recorder, HibernateOrmConfig config, List<PersistenceUnitDescriptorBuildItem> descriptors) {
         PersistenceUnitDescriptor unit = descriptors.get(0).asOutputPersistenceUnitDefinition().getActualHibernateDescriptor();
-        
+
         unit.getProperties().setProperty(AvailableSettings.DIALECT, config.defaultPersistenceUnit.dialect.dialect.orElse(null));
         unit.getProperties().setProperty(AvailableSettings.JPA_TRANSACTION_TYPE, PersistenceUnitTransactionType.JTA.name());
         unit.getProperties().setProperty(AvailableSettings.QUERY_STARTUP_CHECKING, Boolean.FALSE.toString());
@@ -97,9 +97,9 @@ class KeycloakProcessor {
     /**
      * <p>Load the built-in provider factories during build time so we don't spend time looking up them at runtime. By loading
      * providers at this stage we are also able to perform a more dynamic configuration based on the default providers.
-     * 
+     *
      * <p>User-defined providers are going to be loaded at startup</p>
-     * 
+     *
      * @param recorder
      */
     @Record(ExecutionTime.STATIC_INIT)
@@ -119,19 +119,19 @@ class KeycloakProcessor {
                             key -> new HashMap<>())
                             .computeIfAbsent(entry.getKey().getProviderClass(), aClass -> new HashMap<>()).put(factory.getId(),factory.getClass());
                 }
-            }    
+            }
         }
-        
+
         recorder.configSessionFactory(factories, defaultProviders, Environment.isRebuild());
     }
 
     /**
      * <p>Make the build time configuration available at runtime so that the server can run without having to specify some of
      * the properties again.
-     * 
+     *
      * <p>This build step also adds a static call to {@link org.keycloak.cli.ShowConfigCommand#run(Map)} via the recorder
      * so that the configuration can be shown when requested.
-     * 
+     *
      * @param recorder the recorder
      */
     @Record(ExecutionTime.STATIC_INIT)
@@ -217,7 +217,7 @@ class KeycloakProcessor {
 
             factories.put(spi, providers);
         }
-        
+
         return factories;
     }
 
@@ -237,8 +237,8 @@ class KeycloakProcessor {
     }
 
     private void checkProviders(Spi spi,
-            Map<Class<? extends Provider>, Map<String, ProviderFactory>> factoriesMap,
-            Map<Class<? extends Provider>, String> defaultProviders) {
+                                Map<Class<? extends Provider>, Map<String, ProviderFactory>> factoriesMap,
+                                Map<Class<? extends Provider>, String> defaultProviders) {
         String defaultProvider = Config.getProvider(spi.getName());
 
         if (defaultProvider != null) {

--- a/quarkus/runtime/src/main/java/org/keycloak/configuration/KeycloakPropertiesConfigSource.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/configuration/KeycloakPropertiesConfigSource.java
@@ -19,7 +19,6 @@ package org.keycloak.configuration;
 
 import java.io.BufferedReader;
 import java.io.Closeable;
-import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOError;
 import java.io.IOException;
@@ -85,7 +84,7 @@ public abstract class KeycloakPropertiesConfigSource extends PropertiesConfigSou
                 cl = KeycloakPropertiesConfigSource.class.getClassLoader();
             }
             InputStream is;
-            String fileName = "META-INF" + File.separator + KEYCLOAK_PROPERTIES;
+            String fileName = "META-INF/" + KEYCLOAK_PROPERTIES;
             if (cl == null) {
                 is = ClassLoader.getSystemResourceAsStream(fileName);
             } else {

